ra/registration-authority.go

```diff
@@ -381,17 +381,41 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization
 }
 
 // RevokeCertificate terminates trust in the certificate provided.
-func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate) (err error) {
+func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate, reasonCode int, regID *int64) (err error) {
 	serialString := core.SerialToString(cert.SerialNumber)
-	err = ra.CA.RevokeCertificate(serialString, 0)
+	err = ra.CA.RevokeCertificate(serialString, reasonCode)
+
+	state := "Failure"
+	defer func() {
+		// AUDIT[ Revocation Requests ] 4e85d791-09c0-4ab3-a837-d3d67e945134
+		// Needed:
+		//   Serial
+		//   CN
+		//   Revocation reason
+		//   Error (if there was one)
+		revMsg := fmt.Sprintf(
+			"Revocation - State: %s, Serial: %s, CN: %s, DNS Names: %s, Reason: %s",
+			state,
+			serialString,
+			cert.Subject.CommonName,
+			cert.DNSNames,
+			core.RevocationReasons[reasonCode],
+		)
+		// Check regID is set, if not revocation came from the admin-revoker tool
```

Comment:
I like having a separate call as a matter of principle, but after talking to @rolandshoemaker just now it sounds like plumbing the separate call through the RPC layer is going to bloat / delay this change significantly. And as I understand it, we have to have this change deployed by tomorrow, more or less, to pass the audit. Would you consider allowing the "special value" of nil (or some other magic flag value for regId) for now, with a ticket to fix up soon?